packages/abi-coder/src/abi-coder.ts

```diff
@@ -139,7 +139,11 @@ export default class AbiCoder {
     return concat([results, concat(vectorData)]);
   }
 
-  decode(types: ReadonlyArray<JsonAbiFragmentType>, data: BytesLike): DecodedValue[] | undefined {
+  decode(
```

Comment:
It looks a bit weird to pass the logs all the way to the Vec decode method. I was exploring ideas in my mind. To keep this in a more high-level place like the interface and just return a value to know where to place the Vec output parsed from the logs and avoid the throw not being implemented. This would isolate this temp workaround. But don't know if you already tough about this

Reply:
This would remove the logs here and in the vec decoder